pkg/operator/status/status_controller.go

```diff
@@ -124,7 +124,10 @@ func (c StatusSyncer) sync() error {
 	}
 	clusterOperatorObj := originalClusterOperatorObj.DeepCopy()
 
-	if detailedSpec.ManagementState == operatorv1.Unmanaged && !management.IsOperatorAlwaysManaged() {
+	// management.IsOperatorAlwaysManaged() means the operator can't be set to unmanaged state.
+	// so if a cluster admin sets an operator as unmanaged when it can't be unmanaged, then we set the
+	// conditions to unknown, as we are now in an indeterminate state
+	if detailedSpec.ManagementState == operatorv1.Unmanaged && management.IsOperatorAlwaysManaged() {
```

Comment:
This change looks off and the existing code looks right. If are always managed, then entering the block below doesn't make any sense because that operator has no way to be be disabled.  That operator is never unmanaged because the operator simply doesn't accept that spec value.

Reply:
The wording of methods / godoc does not resonate in that way with me, but I'm previously uninitiated to this code repo, so I'll defer and call https://github.com/openshift/library-go/blob/master/pkg/operator/management/management_state.go#L25 in the https://github.com/openshift/cluster-openshift-controller-manager-operator which to date allows for the setting of unmanaged.

Thanks for the clarification